src/shared/containers/challenge-detail/index.jsx

```diff
@@ -299,7 +299,7 @@ const mapStateToProps = (state, props) => ({
   results: state.challenge.results,
   resultsLoadedForChallengeId: state.challenge.resultsLoadedForChallengeId,
   selectedTab: state.challenge.selectedTab || 'details',
-  terms: state.terms.terms,
+  terms: ((state.challenge.details || {}).terms || []).filter(t => t.role === 'Submitter'),
```

Comment:
Good, but this is a wrong thing to do inside `mapStateToProps()`: this method is called now and then, and each time it is called, even if `state.challenge.details` is not changed, the `.filter()` method will create a new array each time; then ReactJS will receive a new array as `terms`, and will re-render the sub-tree where it is used. So, never ever create new objects inside `mapStateToProps()` to pass them into the return value.

The correct way is to make the same, somewhere in the underlying component (like `render()` method, etc.); cause the render method will be re-evaluated only if `mapStateToProps()` has returned a result different from the previous time.

Reply:
@birdofpreyru Thank you for the note. I'll fix it shortly.